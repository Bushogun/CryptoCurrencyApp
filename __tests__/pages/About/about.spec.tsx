import React from 'react';
import { render } from '@testing-library/react';
import About from '@/app/about/page';

describe('About Component', () => {
  it('renders two images with correct attributes', () => {
    const { getByAltText } = render(<About />);
    
    const image1 = getByAltText('Jonathan Soto') as HTMLImageElement;
    const image2 = getByAltText('3D Model JS') as HTMLImageElement;

    expect(image1).toBeInTheDocument();
    expect(image2).toBeInTheDocument();

    const src1 = image1.src;
    const src2 = image2.src;

    expect(src1.includes('untitled.png')).toBeTruthy(); 
    expect(src2.includes('3dmodelJs2.png')).toBeTruthy(); 

    expect(image1).toHaveAttribute('width', '650');
    expect(image1).toHaveAttribute('height', '300');
    expect(image2).toHaveAttribute('width', '450');
    expect(image2).toHaveAttribute('height', '650');
  });
});
