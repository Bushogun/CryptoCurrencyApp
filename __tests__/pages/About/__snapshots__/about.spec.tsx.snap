// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component renders About component correctly 1`] = `
<div>
  <div
    class="about-container"
  >
    <img
      alt="Jonathan Soto"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/_next/image?url=%2Funtitled.png&w=1920&q=75"
      srcset="/_next/image?url=%2Funtitled.png&w=750&q=75 1x, /_next/image?url=%2Funtitled.png&w=1920&q=75 2x"
      style="color: transparent;"
      title="Jonathan Soto"
      width="650"
    />
    <img
      alt="3D Model JS"
      data-nimg="1"
      decoding="async"
      height="650"
      loading="lazy"
      src="/_next/image?url=%2F3dmodelJs2.png&w=1080&q=75"
      srcset="/_next/image?url=%2F3dmodelJs2.png&w=640&q=75 1x, /_next/image?url=%2F3dmodelJs2.png&w=1080&q=75 2x"
      style="color: transparent;"
      title="Jonathan Soto"
      width="450"
    />
  </div>
</div>
`;
